apps/web/app/networking/posts.api.ts

```diff
@@ -0,0 +1,280 @@
+import { createWebApiClient, defineApiConfig } from "./client";
+import type { PostDto } from "@/toolkit/dto/post.dto";
+
+export const CreatePostApi = defineApiConfig((request) => {
+    const postApi = createWebApiClient({
+        routePrefix: "posts",
+        request,
+        options: {
+            withCredentials:false,
+        },
+    });
+    return {
+        async getAllPosts()
+        {
+            // const { data } = await postApi.get<PostDto[]>({
+            //     route:"",
+            // });
+            // return data;
+
+            const data = [
+                {
+                  "id": "post-1235",
+                  "title": "Understanding React Query",
+                  "category": "Web Development",
```

Comment:
Category is going to be an array, because we're supposed to have multiple categories:

```suggestion
                  "categories": ["dova", "zikr", "namaz", "abdest"]
```